components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/bindings/TokenBinder.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.oauth2.token.bindings;
+
+import org.apache.oltu.oauth2.common.exception.OAuthSystemException;
+import org.wso2.carbon.identity.oauth.common.token.bindings.TokenBinderInfo;
+import org.wso2.carbon.identity.oauth2.dto.OAuth2AccessTokenReqDTO;
+
+import java.util.Optional;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * This interface represents the token binder API.
+ */
+public interface TokenBinder extends TokenBinderInfo {
```

Comment:
What is the use case to have 2 interfaces TokenBinder and TokenBinderInfo

Reply:
They are part of the same interface. TokenBinderInfo has only the APIs required for the management console UI generation. TokenBinder provides the APIs for runtime logic.
Due to some class references in some APIs in TokenBinder, we can't put them inside the org.wso2.carbon.identity.oauth. Also, we can't refer  org.wso2.carbon.identity.oauth2 class in org.wso2.carbon.identity.oauth to avoid cyclic dependency.

Reply:
org.wso2.carbon.identity.oauth and org.wso2.carbon.identity.oauth2 are 2 packages in same component, right? In that case, can there be a cyclic dependency?

Reply:
They are two different components actually and org.wso2.carbon.identity.oauth2 component depends on org.wso2.carbon.identity.oauth component.